DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -205,137 +296,96 @@ void TTTrack<T>::setFitParNo(unsigned int nPar) {
 
 template <typename T>
 GlobalVector TTTrack<T>::momentum() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theMomentum;
-  } else
-    return GlobalVector(0.0, 0.0, 0.0);
-}
-
-template <typename T>
-GlobalVector TTTrack<T>::getMomentum(unsigned int npar) const {
-  return momentum();
+  return theMomentum;
 }
 
 template <typename T>
 double TTTrack<T>::rInv() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theRInv;
-  } else
-    return 0.0;
-}
-
-template <typename T>
-double TTTrack<T>::getRInv(unsigned int npar) const {  //backwards compatibility
-
-  return rInv();
+  return theRInv;
 }
 
 template <typename T>
 double TTTrack<T>::tanL() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theTanL;
-  } else
-    return 0.0;
+  return theTanL;
 }
 
 template <typename T>
 double TTTrack<T>::eta() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theEta;
-  } else
-    return 0.0;
+  return theMomentum.eta();
 }
 
 template <typename T>
 double TTTrack<T>::phi() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return thePhi;
-  } else
-    return 0.0;
+  return thePhi;
 }
 
 template <typename T>
 double TTTrack<T>::d0() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theD0;
-  } else
-    return 0.0;
+  return theD0;
 }
 
 template <typename T>
 double TTTrack<T>::z0() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theZ0;
-  } else
-    return 0.0;
+  return theZ0;
 }
 
 template <typename T>
 GlobalPoint TTTrack<T>::POCA() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return thePOCA;
-  } else
-    return GlobalPoint(0.0, 0.0, 0.0);
-}
-
-template <typename T>
-GlobalPoint TTTrack<T>::getPOCA(unsigned int npar) const  //backwards compatibility
-{
-  return POCA();
+  return thePOCA;
 }
 
 /// Chi2
 template <typename T>
 double TTTrack<T>::chi2() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theChi2;
-  } else
-    return 0.0;
+  return theChi2;
 }
 
+/// Chi2z
 template <typename T>
-double TTTrack<T>::getChi2(unsigned int npar) const  //backwards compatibility
-{
-  return chi2();
+double TTTrack<T>::chi2z() const {
+  return theChi2Z;
 }
 
 /// Chi2 reduced
 template <typename T>
 double TTTrack<T>::chi2Red() const {
-  if (NumFitPars == Npars5 || NumFitPars == Npars4) {
-    return theChi2 / (2 * theStubRefs.size() - NumFitPars);
-  } else
-    return 0.0;
+  return theChi2 / (2 * theStubRefs.size() - numFitPars);
```

Comment:
is it worth caching this value (maybe as a transient to maintain the size of the class on disk)?

Reply:
@kpedro88 Wouldn't caching make the class bigger?  I don't understand the sign of this comment?

Reply:
Member variables can be marked as transient so they do not persist, but also do not need to be recalculated every time the accessor is called.